model/ltiProvider/repository/DeliveryLtiProviderRepository.php

```diff
@@ -52,16 +65,18 @@ public function searchByDeliveryExecutionId(string $deliveryExecutionId): LtiPro
             throw new LtiException('This delivery does not contain required lti provider defined');
         }
 
-        return $this->getLtiProvider()->searchById($containerJson['params']['ltiProvider']);
+        return $this->ltiProviderRepository->searchById($containerJson['params']['ltiProvider']);
     }
 
-    private function getDeliveryExecutionService(): DeliveryExecutionService
+    private function findDelivery(string $sourcedId)
     {
-        return $this->getServiceLocator()->get(DeliveryExecutionService::SERVICE_ID);
-    }
+        /** @var DeliveryLookupInterface $provider */
+        foreach ($this->deliveryLookupProviders as $provider) {
+            if ($delivery = $provider->searchBySourcedId($sourcedId)) {
+                return $delivery;
+            }
+        }
 
-    private function getLtiProvider(): LtiProviderRepositoryInterface
-    {
-        return $this->getServiceLocator()->get(LtiProviderService::SERVICE_ID);
+        throw new LtiException('Could not find delivery for provided sourcedId');
```

Comment:
I realized too late this was a breaking change. The endpoint had returned 404 prior to this change in case a delivery execution wasn't found.

Now it returns a 500.

That's because the 404 was set by [`processLisRequest()`](https://github.com/oat-sa/extension-tao-lti-consumer/blob/master/controller/ResultController.php#L106-L107) method in the controller _after_ a successful request parsing. Now, the request parsing actually checks if a delivery or a delivery execution exists, and throws a new exception, which wasn't there before.

Could you please fix that, @augustas?
The endpoint has to return a 404 as it did before in case a delivery execution wasn't found, or, in case of https://github.com/oat-sa/extension-tao-deliver-connect/pull/108, if a sourced_id didn't fit the regex or a delivery wasn't found.

Reply:
This is most interesting.
I did check that behavior on an older instance of TAO, and I get a 404 there in such a scenario.

Here I have two videos.

The first one shows a correct Basic Outcome request handling.

https://user-images.githubusercontent.com/2943256/159439095-060cfd66-4548-41e9-873d-ff08706c97e8.mov

In the second one I replace the sourced_id to an inexistent one, and TAO clearly responds with a 404.

https://user-images.githubusercontent.com/2943256/159439333-79d27660-6e47-4e60-8eb9-23cf40b6e01a.mov

Reply:
Oh, and I forgot to show the error on the TAO 3.x end. Here it is.

```json
{
    "severity": "WARNING",
    "tag": "tao",
    "message": "Delivery execution 'kve_de_https://luoat25exx.lab.taocloud.org/#404_not_found' not found during 'replaceResultRequest' operation processing for ltiProvider with key '9TIY0OK+PtYPmi44tkA=': ",
    "code_file": "/var/www/html/tao/taoLtiConsumer/model/result/ResultService.php",
    "code_line": "74",
    "instance_id": "i-05ab24de31bba3907",
    "stack": "luoat25exx",
    "host_type": "taocloud/ws/delivery"
}
```